### PR TITLE
Enable h2c traffic for http servers.

### PR DESCRIPTION
**Pull request by a-feld**
**Description:** Added support for h2c traffic on http servers generated by confighttp.

This is in preparation for combining gRPC / HTTP ports for OTLP (see #3765 for the complete use case)

**Testing:** The HTTP receive tests have been updated to check for both http/1.1 and h2c traffic when "insecure" mode / non-TLS is used.

### CODE REVIEW

**Comment by a-feld** [comment]
@bogdandrutu @tigrannajaryan @jpkrohling Attempting to break down the gRPC/HTTP port combination work. This PR is a prerequisite to #3765 😄

**Comment by mtwo** [comment]
This needs to be merged before 9/24. @bogdandrutu will review.

File: config/confighttp/confighttp_test.go [hunk 1]
```diff
@@ -420,6 +423,31 @@ func TestHttpReception(t *testing.T) {
 				assert.NoError(t, errRead)
 				assert.Equal(t, "test", string(body))
 			}
+
+			// Check that h2c payloads are handled when serving insecure traffic
+			if tt.tlsClientCreds.Insecure {
```
Comment (bogdandrutu) [inline config/confighttp/confighttp_test.go:428]: Consider to have this as a separate test only for this config. It is not recommended to add this and be run only for one of the "sub-tests".

Comment (a-feld) [reply config/confighttp/confighttp_test.go]: @bogdandrutu done!

**Comment by a-feld** [comment]
@bogdandrutu have you had a chance to review? I'll need to close this by end of day 9/24 (approximately 8 hours from now)

**Comment by alanwest** [comment]
@bogdandrutu Looks like this PR was pretty close. I was not closely involved in this work, but I can get it reopened if it looks good to you.

**Pull request closed without merging**